vault/resource_azure_secret_backend_test.go

```diff
@@ -13,94 +13,118 @@ import (
-				),
+				Check:  resource.ComposeTestCheckFunc(azureUpdatedCheckFuncs...),
 			},
 			{
 				Config: testAzureSecretBackend_updateSubscriptionID(path),
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
-					resource.TestCheckResourceAttr(resourceName, "subscription_id", "11111112-2221-3332-4443-111111111110"),
-					resource.TestCheckResourceAttr(resourceName, "tenant_id", "22222222-3333-4444-5555-333333333333"),
-					resource.TestCheckResourceAttr(resourceName, "client_id", "22222222-3333-4444-5555-444444444444"),
-					resource.TestCheckResourceAttr(resourceName, "client_secret", "098765432109876543214"),
-					resource.TestCheckResourceAttr(resourceName, "environment", "AzurePublicCloud"),
-					resource.TestCheckResourceAttr(resourceName, "use_microsoft_graph_api", "true"),
-				),
+				Check:  resource.ComposeTestCheckFunc(azureUpdatedSubscriptionIDCheckFuncs...),
 			},
 		},
 	})
 }
 
 func TestAzureSecretBackend_remount(t *testing.T) {
+	testutil.SkipTestAcc(t)
+
 	path := acctest.RandomWithPrefix("tf-test-azure")
 	updatedPath := acctest.RandomWithPrefix("tf-test-azure-updated")
 
 	resourceType := "vault_azure_secret_backend"
 	resourceName := resourceType + ".test"
+	azureInitialCheckFuncs := []resource.TestCheckFunc{
+		resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+		resource.TestCheckResourceAttr(resourceName, "subscription_id", "11111111-2222-3333-4444-111111111111"),
+		resource.TestCheckResourceAttr(resourceName, "tenant_id", "11111111-2222-3333-4444-222222222222"),
+		resource.TestCheckResourceAttr(resourceName, "client_id", "11111111-2222-3333-4444-333333333333"),
+		resource.TestCheckResourceAttr(resourceName, "client_secret", "12345678901234567890"),
+		resource.TestCheckResourceAttr(resourceName, "environment", "AzurePublicCloud"),
+	}
+	azureUpdatedCheckFuncs := []resource.TestCheckFunc{
+		resource.TestCheckResourceAttr(resourceName, consts.FieldPath, updatedPath),
+		resource.TestCheckResourceAttr(resourceName, "subscription_id", "11111111-2222-3333-4444-111111111111"),
+		resource.TestCheckResourceAttr(resourceName, "tenant_id", "11111111-2222-3333-4444-222222222222"),
+		resource.TestCheckResourceAttr(resourceName, "client_id", "11111111-2222-3333-4444-333333333333"),
+		resource.TestCheckResourceAttr(resourceName, "client_secret", "12345678901234567890"),
+		resource.TestCheckResourceAttr(resourceName, "environment", "AzurePublicCloud"),
+	}
+
+	skipMSGraphCheck := provider.IsAPISupported(testProvider.Meta(), provider.VaultVersion112)
+	if !skipMSGraphCheck {
+		azureInitialCheckFuncs = append(azureInitialCheckFuncs,
+			resource.TestCheckResourceAttr(resourceName, "use_microsoft_graph_api", "false"))
+		azureUpdatedCheckFuncs = append(azureUpdatedCheckFuncs,
+			resource.TestCheckResourceAttr(resourceName, "use_microsoft_graph_api", "false"))
+	}
+
```

Comment:
We can replace this block of code with the following in order to remove some of the copy-paste code: 

```
commonChecks := []resource.TestCheckFunc{
		resource.TestCheckResourceAttr(resourceName, "subscription_id", "11111111-2222-3333-4444-111111111111"),
		resource.TestCheckResourceAttr(resourceName, "tenant_id", "11111111-2222-3333-4444-222222222222"),
		resource.TestCheckResourceAttr(resourceName, "client_id", "11111111-2222-3333-4444-333333333333"),
		resource.TestCheckResourceAttr(resourceName, "client_secret", "12345678901234567890"),
		resource.TestCheckResourceAttr(resourceName, "environment", "AzurePublicCloud"),
	}

	skipMSGraphCheck := provider.IsAPISupported(testProvider.Meta(), provider.VaultVersion112)
	if !skipMSGraphCheck {
		commonChecks = append(commonChecks,
			resource.TestCheckResourceAttr(resourceName, "use_microsoft_graph_api", "false"))
	}

	azureInitialCheckFuncs := append(commonChecks,
		resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
	)

	azureUpdatedCheckFuncs := append(commonChecks,
		resource.TestCheckResourceAttr(resourceName, consts.FieldPath, updatedPath),
	)
```

Reply:
Got it! Makes sense! Thank you 👍